### ISSUE TITLE
Multi-level GP not outputting expected errors

### ISSUE THREAD

**Issue by HarryTWhite** [EXA-UQ/EXAUQ-Toolbox#330]
### Motivation
-----------------

@JSalter90 

**TLDR:** Could you reply to this with a code snippet set up that you used and expected MLGP to give you an error but didn't - alongside the error you expected to see.

From meeting (24/10/2024) one of the researchers mentioned they experienced/encountered a possible bug with high dimensions that meant the MLGP didn't output the expected errors.  This is important to note as it could be that there are issues at higher dimensions where either `mogp` or part of `MultiLevelGaussianProcess` is not performing correctly yet not throwing errors out where it should. Or, these error messages are being suppressed where they shouldn't.

### Potential Silencing 
--------------------

There are 2 potential silencing issues that we should rule out first:
1) There could be a warning but in the tutorials we use: 
```python
# Suppress warnings that arise from mogp_emulator
import warnings
warnings.filterwarnings("ignore")
```
to prevent things like the deprecation warning from NumPy showing etc. 

2) It could be the `@suppress_print` decorator which is used to silence `mogp` from certain print statements is also silencing this warning as there isn't an inbuilt logging system into `mogp`. 

### Acceptance Criteria
--------------------------

To resolve this issue, it should be that MLGP throws up the expected error for high dimensions and that other potential errors are not being silenced either by the warnings or by  `@suppress_print`. It may be that further testing is involved, or it could be that there are incorrectly running functions somewhere that need to be investigated. This issue will be updated once we can be more certain.

**Comment by JSalter90** [comment]
@mbjohns I've revisited this issue under the current version of the code. I get the same issue (tiny variance on level 2) running the above code, but the mogp emulator does fit, and using a different seed for the level 2 data does change things. Issue is parameter estimation is poorly defined when number of points < number of parameters, so sometimes gives results that look clearly wrong, sometimes gives results that aren't obviously wrong.

I think the simple solution is to add a warning message if mlgp.fit(training_data) is trying to fit to too small a dataset. Something like 'Warning: number of parameters being estimated is larger than the size of the training data for level {lvl}, parameter estimation are likely poor."

**Comment by JSalter90** [comment]
Closed by #404
